### PR TITLE
test: test_mcuboot: enable test on NXP MCUX platforms

### PR DESCRIPTION
**Pull request by hakehuang**
NXP MCUX has no issue with #58080, so enable tests for NXP

### CODE REVIEW

File: tests/boot/test_mcuboot/testcase.yaml
Comment (butok) [inline tests/boot/test_mcuboot/testcase.yaml:19]: Do we want to disable nrf52840dk_nrf52840?

**Comment by butok** [comment]
BTW:
As we may see, these twister .yml files are not updated. 
For future, we should avoid hiding issues via "skip: true", and better report failed tests to warn that an issue is still present and  should be fixed asap.

**Comment by nordicjm** [comment]
In this case, the `skip: true` is not because of a test issue, it is an issue with gcc's binutils and not something that can be fixed without applying a patch. The issue itself can impact any test almost at random depending upon how large a specific section is

**Comment by butok** [comment]
> In this case, the `skip: true` is not because of a test issue, it is an issue with gcc's binutils and not something that can be fixed without applying a patch. The issue itself can impact any test almost at random depending upon how large a specific section is

It can be marked as a known issue (adding attention to it), but not by disabling/hiding the test for all.

**Comment by nordicjm** [comment]
> > In this case, the `skip: true` is not because of a test issue, it is an issue with gcc's binutils and not something that can be fixed without applying a patch. The issue itself can impact any test almost at random depending upon how large a specific section is
> 
> It can be marked as a known issue (adding attention to it), but not by disabling/hiding the test for all.

Then every PR run would fail and no PRs could be merged.

**Comment by butok** [comment]
> > > In this case, the `skip: true` is not because of a test issue, it is an issue with gcc's binutils and not something that can be fixed without applying a patch. The issue itself can impact any test almost at random depending upon how large a specific section is
> > 
> > 
> > It can be marked as a known issue (adding attention to it), but not by disabling/hiding the test for all.
> 
> Then every PR run would fail and no PRs could be merged.

The known-issues should be in a Quarantine list (result ignored), not disabled.
Otherwise, it is not possible to get a real test picture.

**Comment by danieldegrasse** [comment]
> In this case, the skip: true is not because of a test issue, it is an issue with gcc's binutils and not something that can be fixed without applying a patch. The issue itself can impact any test almost at random depending upon how large a specific section is

While not a solution for every platform (especially those with strict MPU alignment restrictions), #60934 added `CONFIG_BUILD_ALIGN_LMA`, which enables the test to pass for platforms where this issue shows up, at the cost of some wasted ROM space in the output image.

We might be able to leverage this Kconfig to enable the test?

File: tests/boot/test_mcuboot/testcase.yaml
Comment (hakehuang) [reply tests/boot/test_mcuboot/testcase.yaml:19]: Nordic guys add https://github.com/zephyrproject-rtos/zephyr/issues/58080 for their platform, so to enable us, I have to skip theirs for now

**Comment by hakehuang** [comment]
@butok  RT1010/RT1015/RT1040 do not support mcuboot, and this case has build failure form those platform as well as lpc55s28 platform.

**Comment by butok** [comment]
> @butok according to the CI https://github.com/zephyrproject-rtos/zephyr/actions/runs/6622116782/job/17987172186?pr=64250, 8 platforms not support mcuboot. RT1010/RT1015/RT1040 LPC55S06/LPC55S16/LPC55S69_cpu0 mimxrt1170_evk_cm4/mimxrt1160_evk_cm4

We have built (but did not notice run results) :
INFO    -  9/20 mimxrt1024_evk            tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 10/20 mimxrt685_evk_cm33        tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 11/20 mimxrt1020_evk            tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 12/20 mimxrt595_evk_cm33        tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 13/20 mimxrt1050_evk            tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 14/20 frdm_k64f                 tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 15/20 mimxrt1064_evk            tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 16/20 mimxrt1060_evk            tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 17/20 mimxrt1160_evk_cm4        tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 18/20 b_u585i_iot02a            tests/boot/test_mcuboot/bootloader.mcuboot.assert  PASSED (build)
INFO    - 19/20 mimxrt1160_evk_cm7        tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)
INFO    - 20/20 mimxrt1170_evk_cm7        tests/boot/test_mcuboot/bootloader.mcuboot         PASSED (build)

- mimxrt1170_evk_cm4/mimxrt1160_evk_cm4 we can ignore, as CM7 is passed.
- LPC55s69/36/16/06 should be passed after all pending PRs are approved (waiting for a review several weeks).

So, TBD to cover:
- RT1010/RT1015/RT1040
- lpc55s28

**Review by butok — DISMISSED** [review]

**Comment by hakehuang** [comment]
> * LPC55s69/36/16/06 should be passed after all pending PRs are approved (waiting for a review several weeks).

@butok can you list the PR in this commits, and can remove the draft PR after your PR merged. Thanks.

**Comment by hakehuang** [comment]
> We have built (but did not notice run results) :

> INFO - 10/20 mimxrt685_evk_cm33 tests/boot/test_mcuboot/bootloader.mcuboot PASSED (build)
> INFO - 12/20 mimxrt595_evk_cm33 tests/boot/test_mcuboot/bootloader.mcuboot PASSED (build)
> INFO - 13/20 mimxrt1050_evk tests/boot/test_mcuboot/bootloader.mcuboot PASSED (build)
> INFO - 14/20 frdm_k64f tests/boot/test_mcuboot/bootloader.mcuboot PASSED (build)
> INFO - 15/20 mimxrt1064_evk tests/boot/test_mcuboot/bootloader.mcuboot PASSED (build)
> INFO - 16/20 mimxrt1060_evk tests/boot/test_mcuboot/bootloader.mcuboot PASSED (build)
> INFO - 19/20 mimxrt1160_evk_cm7 tests/boot/test_mcuboot/bootloader.mcuboot PASSED (build)
> INFO - 20/20 mimxrt1170_evk_cm7 tests/boot/test_mcuboot/bootloader.mcuboot PASSED (build)

if I enable the CI test with this PR, the test on above platfrom is OK.  i will add CI for systebuild support in imxrt102x series. 
@butok do you have chance to setup a run/test environment for boards you have? I do not have this board 'b_u585i_iot02a '

**Comment by butok** [comment]
> I do not have this board 'b_u585i_iot02a '

This is the ST board: https://www.st.com/en/evaluation-tools/b-u585i-iot02a.html
We do not support it ;)

**Comment by butok** [comment]
> @butok do you have chance to setup a run/test environment for boards you have? 

I do not have a dedicated QA and automated run/test environment. That is why we rely on you.
I can do testing manually, one by one.

**Comment by hakehuang** [comment]
> I do not have a dedicated QA and automated run/test environment. That is why we rely on you.
> I can do testing manually, one by one.

ok, please list your pending PRs here, I can try to verify as well

**Comment by butok** [comment]
> ok, please list your pending PRs here, I can try to verify as well

For LPC testing better to wait, 3 pending PR for Zephyr-rtos. 
Also, as I see, MCUBoot has not updated to the v2.0 yet, so LPC configuration files are missed.

The proposal:
- now, you can enable platforms that are passed for you. 
- later, I will enable the rest of failed platforms (LPC and RT10xx).

File: tests/boot/test_mcuboot/testcase.yaml
Comment (nordicjm) [inline tests/boot/test_mcuboot/testcase.yaml:30]: uncomment these boards

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: tests/boot/test_mcuboot/testcase.yaml
Comment (hakehuang) [reply tests/boot/test_mcuboot/testcase.yaml:30]: ok

**Review by butok — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**
